Testscripts/Linux/ETHTOOL-OFFLOADING-SETTING.sh

```diff
@@ -0,0 +1,268 @@
+            LogMsg "Set ${offload_features[$key]} on"
+            result=$(ethtool -K $synthetic_interface_vm_1 ${offload_features[$key]} on 2>&1)
+            if [[ "$result" = *"Could not change any device features"* ]]; then
+                LogErr "$result"
+                LogErr "The kernel doesn't support this feature"
+            fi
+        fi
+    done
+
+    tx_value=$(cat /sys/class/net/"${vf_interface_vm_1}"/statistics/tx_packets)
+    LogMsg "TX value before disabling and enabling the VF: $tx_value"
+
+    # Get the VF name from VM2
+    cmd_to_send="ip addr | grep \"$static_IP_2\" | awk '{print \$NF}'"
+    synthetic_interface_vm_2=$(ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$static_IP_2" "$cmd_to_send")
+    if [[ $DISTRO_VERSION =~ ^6\. ]]; then
+        synthetic_MAC_command="ip link show "${synthetic_interface_vm_2}" | grep ether | awk '{print \$2}'"
+        synthetic_MAC=$(ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$static_IP_2" "$synthetic_MAC_command")
+        cmd_to_send="grep -il ${synthetic_MAC} /sys/class/net/*/address | grep -v "${synthetic_interface_vm_2}" | sed 's/\// /g' | awk '{print \$4}'"
+    else
+        cmd_to_send="find /sys/devices/* -name "*${synthetic_interface_vm_2}" | grep pci | sed 's/\// /g' | awk '{print \$12}'"
+    fi
+    vf_interface_vm_2=$(ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$static_IP_2" "$cmd_to_send")
+
+    rx_value=$(ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$static_IP_2" cat /sys/class/net/"${vf_interface_vm_2}"/statistics/rx_packets)
+    LogMsg "RX value before disabling and enabling the VF: $rx_value"
+
+    if ! DisableEnablePCI "SR-IOV"; then
+        LogErr "Could not disable and reenable pci device."
+        SetTestStateFailed
+        exit 0
+    fi
+
+    for key in ${!offload_features[*]}
+    do
+        on_or_off=$(ethtool -k $synthetic_interface_vm_1 2>&1 | grep $key | awk {'print $2'})
+        fixed_or_not=$(ethtool -k $synthetic_interface_vm_1 2>&1 | grep $key | awk {'print $3'})
+        LogMsg "The final status of $key for $synthetic_interface_vm_1: $on_or_off $fixed_or_not"
+    done
+
+    tx_value_final=$(cat /sys/class/net/"${vf_interface_vm_1}"/statistics/tx_packets)
+    LogMsg "TX value after disabling and enabling the VF: $tx_value_final"
+
+    rx_value_final=$(ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no "$remote_user"@"$static_IP_2" cat /sys/class/net/"${vf_interface_vm_2}"/statistics/rx_packets)
+    LogMsg "RX value after disabling and enabling the VF: $rx_value_final"
+
+    if [[ "$tx_value_final" -lt "$tx_value" || "$rx_value_final" -lt "$rx_value" ]]; then
+        LogErr "TX or RX packets drops after disabling and enabling the VF"
+        SetTestStateFailed
+        exit 0
+    fi
+
+    # The 7000000 is an experience value, not an accurate value
```

Comment:
@haiyangz  Please advise about expected final value. @xian123 put 7000000. Can you confirm if this is ok?

Reply:
It's the same for @LiliDeng @Tong-Xu to email out to @haiyangz for this confirmation.

Reply:
I think we can just remove this logic to check tx/rx_value_final.  As we have checked the main functionality before this code, and make little sense to me for checking an experienced value '7000000' here.  This may introduce root causes of unstable.  More info is that, we may introduce delay for kernel at any vmbus channels, this number may not make sense.  (even this is a SRIOV scenario)

Reply:
Updated the code and re-run the test PASS.